frontend/public/components/modals/alert-routing-modal.tsx

```diff
@@ -18,23 +33,28 @@ export const AlertRoutingModal: React.FC<AlertRoutingModalProps> = ({
 
   const submit = (event): void => {
     event.preventDefault();
-    const configObj = _.cloneDeep(config);
-    _.set(
-      configObj,
-      ['route', 'group_by'],
-      event.target.elements['input-group-by'].value.split(','),
-    );
-    _.set(configObj, ['route', 'group_wait'], event.target.elements['input-group-wait'].value);
-    _.set(
-      configObj,
-      ['route', 'group_interval'],
-      event.target.elements['input-group-interval'].value,
-    );
-    _.set(
-      configObj,
-      ['route', 'repeat_interval'],
-      event.target.elements['input-repeat-interval'].value,
-    );
+    const configObj = !_.isEmpty(config) ? _.cloneDeep(config) : {};
+
+    const groupByNew = event.target.elements['input-group-by'].value.replace(/\s+/g, '');
+    const groupWaitNew = event.target.elements['input-group-wait'].value;
+    const groupIntervalNew = event.target.elements['input-group-interval'].value;
+    const repeatIntervalNew = event.target.elements['input-repeat-interval'].value;
+
+    const groupByOld = _.get(configObj, ['route', 'group_by'], []);
+    const groupWaitOld = _.get(configObj, ['route', 'group_wait'], '');
+    const groupIntervalOld = _.get(configObj, ['route', 'group_interval'], '');
+    const repeatIntervalOld = _.get(configObj, ['route', 'repeat_interval'], '');
+
+    if (groupByNew !== groupByOld.join(',')) {
+      if (_.isEmpty(groupByNew)) {
+        _.unset(configObj, ['route', 'group_by']);
+      } else {
+        _.set(configObj, ['route', 'group_by'], groupByNew.split(','));
+      }
+    }
```

Comment:
If `updateAlertRoutingProperty` used `!_.isEqual` instead of `!==`, I think you wouldn't need this special case for `group_by` and could just pass `groupByNew.split(',')` into `updateAlertRoutingProperty` instead.

Reply:
you are correct, fixed